boa/profiling.py

```diff
@@ -145,17 +144,24 @@ def by_pc(self):
     def by_line(self):
         ret = {}
         source_map = self.contract.source_map["pc_raw_ast_map"]
-        current_line = None
         seen = set()
         for pc in self.computation.code._trace:
-            if (node := source_map.get(pc)) is not None:
-                current_line = node.lineno
+            if pc in seen:
+                # TODO: this is a kludge, it prevents lines from being
+                # over-represented when they appear in loops. but we should
+                # probably either not have this guard, or actually count
```

Comment:
Shouldn't this work given we are calling `merge`?

Reply:
no, we skip `merge` after the first time we see a pc.